lib/trino-parquet/src/main/java/io/trino/parquet/writer/valuewriter/BigintValueWriter.java

```diff
@@ -39,6 +42,7 @@ public void write(Block block)
                 long value = type.getLong(block, i);
                 getValueWriter().writeLong(value);
                 getStatistics().updateStats(value);
+                getBloomFilter().ifPresent(filter -> filter.insertHash(filter.hash(((Number) value).longValue())));
```

Comment:
With dictionary page size=1 (+some extra modifications, so not directly comparable to the above benchmark) bloom filters are twice as slow, possibly because we process twice as much data:

## Binary

```
Benchmark                             (bloomFilterType)  (positionLength)                     (type)   Mode  Cnt    Score    Error  Units
BenchmarkBinaryColumnWriter.write                  NONE    VARIABLE_0_100                  UNBOUNDED  thrpt   20  571.614 ± 21.855  ops/s
BenchmarkBinaryColumnWriter.write                  NONE    VARIABLE_0_100  VARCHAR_ASCII_BOUND_EXACT  thrpt   20  563.650 ± 42.195  ops/s
BenchmarkBinaryColumnWriter.write                  NONE          FIXED_10                  UNBOUNDED  thrpt   20  688.586 ± 16.662  ops/s
BenchmarkBinaryColumnWriter.write                  NONE          FIXED_10  VARCHAR_ASCII_BOUND_EXACT  thrpt   20  683.202 ± 15.252  ops/s
BenchmarkBinaryColumnWriter.write  DEFAULT_BLOOM_FILTER    VARIABLE_0_100                  UNBOUNDED  thrpt   20  280.634 ±  4.330  ops/s
BenchmarkBinaryColumnWriter.write  DEFAULT_BLOOM_FILTER    VARIABLE_0_100  VARCHAR_ASCII_BOUND_EXACT  thrpt   20  278.466 ±  2.464  ops/s
BenchmarkBinaryColumnWriter.write  DEFAULT_BLOOM_FILTER          FIXED_10                  UNBOUNDED  thrpt   20  379.657 ±  3.027  ops/s
BenchmarkBinaryColumnWriter.write  DEFAULT_BLOOM_FILTER          FIXED_10  VARCHAR_ASCII_BOUND_EXACT  thrpt   20  388.284 ±  6.712  ops/s
```

## Long

```
Benchmark                        (bitWidth)     (bloomFilterType)   Mode  Cnt     Score    Error  Units
BenchmarkLongColumnWriter.write           0                  NONE  thrpt   20  1985.634 ± 52.654  ops/s
BenchmarkLongColumnWriter.write           0  DEFAULT_BLOOM_FILTER  thrpt   20   906.034 ± 58.090  ops/s
BenchmarkLongColumnWriter.write          20                  NONE  thrpt   20  2036.688 ± 38.065  ops/s
BenchmarkLongColumnWriter.write          20  DEFAULT_BLOOM_FILTER  thrpt   20   898.417 ± 49.697  ops/s
BenchmarkLongColumnWriter.write          35                  NONE  thrpt   20  1933.658 ± 48.521  ops/s
BenchmarkLongColumnWriter.write          35  DEFAULT_BLOOM_FILTER  thrpt   20   877.394 ± 83.834  ops/s
BenchmarkLongColumnWriter.write          50                  NONE  thrpt   20  1928.422 ± 54.861  ops/s
BenchmarkLongColumnWriter.write          50  DEFAULT_BLOOM_FILTER  thrpt   20   973.334 ± 11.228  ops/s
BenchmarkLongColumnWriter.write          64                  NONE  thrpt   20  1943.113 ± 61.978  ops/s
BenchmarkLongColumnWriter.write          64  DEFAULT_BLOOM_FILTER  thrpt   20   928.034 ± 38.232  ops/s
```

Reply:
Given that the only difference between the benchmark runs should be the statement `filter.insertHash(filter.hash(v));`, this seems to be a limitation of the API parquet-mr offers. Possibly we could compute the xxhash of each block using Trino, and then only need to do `filter.insertHash`. Though, it's unclear if running xxhash using Trino would be any faster.